### PR TITLE
Improving documentation

### PR DESCRIPTION
**Pull request by FauziAkram**
Improving documentation.

### CODE REVIEW

File: docs/nnue.md [hunk 1]
```diff
@@ -1494,13 +1494,13 @@ int32_t* linear_sparse_input(
     constexpr int num_tiles = layer.num_outputs / tile_size;
 
     // We need to find out the indices of the input values that are non-zero
-    // We'll use a lookup table approach. Overallocate 16 elements
+    // We'll use a lookup table approach. Over allocate 16 elements
```
Comment (Sopel97) [inline docs/nnue.md]: "overallocate" is correct here

File: docs/nnue.md [hunk 2]
```diff
@@ -1756,19 +1756,19 @@ int32_t* linear_sparse_input(
 
 ##### Helping the odds
 
-The math of `x^4` assumes uniform distribution of non-zero inputs. We, however, help it a little bit by reordering the weights such that values that are more likely to be non-zero are grouped together (say, at the beginning). This can be performed empirically. This is a minor (~2%), but essentially free, speedup!
+The math of `x^4` assumes a uniform distribution of non-zero inputs. We, however, help it a little bit by reordering the weights such that values that are more likely to be non-zero are grouped together (say, at the beginning). This can be performed empirically. This is a minor (~2%), but essentially free, speedup!
```
Comment (Sopel97) [inline docs/nnue.md:1759]: the addition of "a" here is questionable, but ok

File: docs/nnue.md [hunk 3]
```diff
@@ -1890,7 +1890,7 @@ int8_t* crelu16(,
             _mm256_permute4x64_epi64(
                 // clamp from below
                 _mm256_max_epi8(
-                    // packs saturates to 127, so we only need to clamp from below
+                    // packs saturate to 127, so we only need to clamp from below
```
Comment (Sopel97) [inline docs/nnue.md]: "saturates" was correct, "packs" is singular

File: docs/nnue.md [hunk 4]
```diff
@@ -1999,14 +1999,14 @@ int8_t* quantmoid4(
         v0 = _mm256_subs_epu16(cst_127_epi16, _mm256_abs_epi16(v0));
         v1 = _mm256_subs_epu16(cst_127_epi16, _mm256_abs_epi16(v1));
 
-        // Since we use mulhi later we have to prepare the input such that
+        // Since we use multi later we have to prepare the input such that
```
Comment (Sopel97) [inline docs/nnue.md]: wrong

File: docs/nnue.md [hunk 5]
```diff
@@ -2345,7 +2345,7 @@ if __name__ == '__main__':
 
 ### Using custom CUDA kernels
 
-How to run our own kernel? Don't we need a complicated setup with the CUDA compiler and all that? CuPy to the rescue. CuPy is a python library that allows easy creation of CUDA kernels using plain python strings containing the CUDA code. CuPy handles compilation and everything else for us. For example:
+How to run our own kernel? Don't we need a complicated setup with the CUDA compiler and all that? CuPy to the rescue. CuPy is a python library that allows the easy creation of CUDA kernels using plain python strings containing the CUDA code. CuPy handles compilation and everything else for us. For example:
```
Comment (Sopel97) [inline docs/nnue.md]: imo reads easier and is clearer without the "the"

**Review by Sopel97 — CHANGES_REQUESTED** [review]
thanks for taking the time to proof-read this

File: docs/nnue.md (hunk 5 above)
Comment (FauziAkram) [reply docs/nnue.md]: Reverted those ones.

**Comment by FauziAkram** [comment]
@Sopel97 You are welcome.
Always ready to help.
Reverted those.

**Pull request merged**